rest-api/src/main/scala/za/co/absa/enceladus/rest_api/services/v3/SchemaServiceV3.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.rest_api.services.v3
+
+import org.apache.spark.sql.types.StructType
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import za.co.absa.enceladus.model.{Schema, UsedIn, Validation}
+import za.co.absa.enceladus.rest_api.repositories.{DatasetMongoRepository, MappingTableMongoRepository, SchemaMongoRepository}
+import za.co.absa.enceladus.rest_api.services.{SchemaService, VersionedModelService}
+import za.co.absa.enceladus.rest_api.utils.converters.SparkMenasSchemaConvertor
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
+import scala.concurrent.Future
+
+@Service
+class SchemaServiceV3 @Autowired()(schemaMongoRepository: SchemaMongoRepository,
+                                   mappingTableMongoRepository: MappingTableMongoRepository,
+                                   datasetMongoRepository: DatasetMongoRepository,
+                                   sparkMenasConvertor: SparkMenasSchemaConvertor)
+  extends VersionedModelService(schemaMongoRepository) {
+
+  // same as v2
+  override def getUsedIn(name: String, version: Option[Int]): Future[UsedIn] = {
```

Comment:
I am thinking why not just add the non-existing methods in the old Services and not need to copy all these methods. After all, services are supposed to be reusable

Reply:
Solved together with https://github.com/AbsaOSS/enceladus/pull/2052#discussion_r861561310 below